### PR TITLE
Log all image deletion errors instead of just the last one

### PR DESCRIPTION
**Pull request by ronnielai**
#27169

### CODE REVIEW

File: pkg/kubelet/image_manager.go [hunk 1]
```diff
@@ -269,7 +269,7 @@ func (im *realImageManager) freeSpace(bytesToFree int64, freeTime time.Time) (in
 	sort.Sort(byLastUsedAndDetected(images))
 
 	// Delete unused images until we've freed up enough space.
-	var lastErr error
+	deletionErrors := make(map[string]error)
```
Comment (yujuhong) [inline pkg/kubelet/image_manager.go]: There is a pkg/util/errors[https://github.com/kubernetes/kubernetes/blob/master/pkg/util/errors/errors.go] package if you are interested in aggregating al errors as a single error.

File: pkg/kubelet/image_manager.go [hunk 2]
```diff
@@ -302,7 +302,10 @@ func (im *realImageManager) freeSpace(bytesToFree int64, freeTime time.Time) (in
 		}
 	}
 
-	return spaceFreed, lastErr
+	if len(deletionErrors) > 0 {
+		return spaceFreed, fmt.Errorf("Wated to free %d, but freed %d space with errors in image deletion: %v", bytesToFree, spaceFreed, deletionErrors)
```
Comment (yujuhong) [inline pkg/kubelet/image_manager.go]: nit: s/Wated/wanted

typo and also error message starts with a lower case letter.

**Comment by yujuhong** [comment]
@dchen1107 just taking over some of your load.

File: pkg/kubelet/image_manager.go (hunk 1 above)
Comment (ronnielai) [inline pkg/kubelet/image_manager.go]: TIL and fixed.

File: pkg/kubelet/image_manager.go (hunk 2 above)
Comment (ronnielai) [inline pkg/kubelet/image_manager.go]: Fixed.

**Pull request merged**